tensorpipe/test/CMakeLists.txt

```diff
@@ -82,27 +93,40 @@ if(TP_USE_CUDA)
     core/pipe_cuda_test.cc
     )
 
-  list(APPEND TP_TEST_SRCS
-    channel/cuda_xth/cuda_xth_test.cc
-    channel/cuda_basic/cuda_basic_test.cc
-    )
+  if(TP_ENABLE_HIP_XTH OR TP_USE_CUDA)
+    list(APPEND TP_TEST_SRCS
+      channel/cuda_xth/cuda_xth_test.cc
+      channel/cuda_basic/cuda_basic_test.cc
+      )
+  endif()
 
-  if(TP_ENABLE_CUDA_IPC)
+  if(TP_ENABLE_CUDA_IPC OR TP_ENABLE_HIP_IPC)
     list(APPEND TP_TEST_SRCS
       channel/cuda_ipc/cuda_ipc_test.cc
       )
   endif()
 
-  list(APPEND TP_TEST_SRCS
-    channel/cuda_gdr/cuda_gdr_test.cc
-    )
+  if((TP_ENABLE_CUDA_GDR AND TP_USE_CUDA) OR (TP_ENABLE_HIP_GDR AND TP_USE_ROCM))
```

Comment:
@beauby do you know why we don't need this if-clause in the existing code?

Reply:
At the moment we manually exclude Gdr tests from our CI, but it would not hurt to avoid building the tests if support for CudaGdr is not built.